lib/compilers/zig.js

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2018, Marc Tiehuis
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+const BaseCompiler = require('../base-compiler'),
+    _ = require('underscore'),
+    path = require('path');
+
+class ZigCompiler extends BaseCompiler {
+    constructor(info, env) {
+        super(info, env);
+        this.compiler.supportsIntel = true;
+    }
+
+    preProcess(source) {
+        source += '\n';
+        source += 'extern fn zig_panic() noreturn;\n';
+        source += 'pub inline fn panic(msg: []const u8, error_return_trace: ' +
+                  '?*@import("builtin").StackTrace) noreturn {\n';
+        source += '    zig_panic();\n';
+        source += '}\n';
+
+        return source;
+    }
+
+    optionsForFilter(filters, outputFilename, userOptions) {
+        let options = [
+            'build-obj',
+            '--cache-dir', path.dirname(outputFilename),
+            '--output', this.filename(outputFilename)
+        ];
+
+        let userRequestedEmit = _.any(userOptions, opt => opt.indexOf("--emit") > -1);
```

Comment:
If you move this line inside the if, we save the check to only when it will actually be needed :)